Source/WebKit/WebProcess/WebPage/RemoteLayerTree/RemoteLayerTreeDrawingArea.mm

```diff
@@ -456,6 +456,9 @@
     while (!monitorsToNotify.isEmpty())
         monitorsToNotify.takeAny()->triggerDisplayDidRefresh();
     m_displayRefreshMonitorsToNotify = nullptr;
+
+    if (!m_updateRenderingTimer.isActive())
```

Comment:
Could `triggerDisplayDidRefresh` just return a bool to say if it's going to do anything, rather than checking the timer?

Reply:
I don't think we easily can, since we're just triggering `displayDidRefresh` on some number of display refresh monitors. One of those might have `RenderingUpdateScheduler` as a client, and that might ask this same drawing area to trigger a rendering update. I don't think we can hardcode the 'did trigger rendering update' into the `DisplayRefreshMonitor` API.

It seems plausible that we could remove some of the abstraction layers here, but that's a separate task (that I am now intrigued with).

Reply:
Following up, the WIP PR #14327 would make this possible.